src/cgutils.cpp

```diff
@@ -3163,6 +3163,45 @@ static Value *box_union(jl_codectx_t &ctx, const jl_cgval_t &vinfo, const SmallB
     return box_merge;
 }
 
+static Function *mangleIntrinsic(IntrinsicInst *call)
+{
+    Intrinsic::ID ID = call->getIntrinsicID();
+    auto nargs = call->arg_size();
+    SmallVector<Value*, 8> args(nargs);
+    SmallVector<Type*, 8> argTys(nargs);
+    for (unsigned i = 0; i < nargs; i++) {
+        auto arg = call->getArgOperand(i);
+        args[i] = arg;
+        argTys[i] = args[i]->getType();
+    }
```

Comment:
You should not be making an array of args if you don't need it
```suggestion
    SmallVector<Type*, 8> argTys(nargs);
    for (unsigned i = 0; i < nargs; i++) {
        auto argi = call->getArgOperand(i);
        argTys[i] = argi->getType();
    }
```

Reply:
It is also slightly extraneous to iterate up to `nargs`, when you are going to drop everything after `oldfType->getNumParams()`